### DISCUSSION TITLE
Why setting `shouldDirty` to `false` when calling `setValue`

### DISCUSSION THREAD

**Discussion by delijah** [react-hook-form/react-hook-form#9350] [Q&A]
Hello

We are asking ourselves, why the `shouldDirty` option on the `setValue` function is set to `false` by default. We are pretty sure we want to keep our dirty states in sync all the time, so we basically always would have to set the `shouldDirty` option to `true`.

What is the idea behind setting it to `false` by default? We feel like it's pretty dangerous (error prone) to have out of sync dirty states, `dirtyFields` etc.

Thanks for clarification!

**Comment by bluebill1049** [comment]
The original idea was it's programmatically updated input value instead of the user and potentially performance less computation within the library by default, but we can consider changing that in the next major version.

**Comment by delijah** [comment]
Ok. So i'll open a feature request soon!

**Comment by delijah** [comment]
Done: https://github.com/react-hook-form/react-hook-form/issues/11068
